oauth2/src/main/java/com/okta/spring/boot/oauth/config/OktaOAuth2Properties.java

```diff
@@ -179,4 +192,47 @@ public void validate(Object target, Errors errors) {
                     errors.rejectValue("issuer", res.getMessage()));
         }
     }
+
+    public static class Proxy {
+
+        private String host;
+
+        private int port;
+
+        private String username;
+
+        private String password;
+
+        public String getHost() {
+            return host;
+        }
+
+        public void setHost(String host) {
+            this.host = host;
+        }
+
+        public int getPort() {
+            return port;
+        }
+
+        public void setPort(int port) {
+            this.port = port;
+        }
```

Comment:
nit: consider validating the `Proxy` object above you have an if statement that both `host` and `port` either both required or both should be missing, but there is no validation, if one of them is missing